internal/clients/azure/azure.go

```diff
@@ -0,0 +1,65 @@
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/kubelogin/pkg/token"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/rest"
+)
+
+// Credentials Secret content is a json whose keys are below.
+const (
+	CredentialsKeyClientID       = "clientId"
+	CredentialsKeyClientSecret   = "clientSecret"
+	CredentialsKeyTenantID       = "tenantId"
+	CredentialsKeyClientCert     = "clientCertificate"
+	CredentialsKeyClientCertPass = "clientCertificatePassword"
+)
+
+// WrapRESTConfig configures the supplied REST config to use OAuth2 bearer
+// tokens fetched using the supplied Azure Credentials.
+func WrapRESTConfig(_ context.Context, rc *rest.Config, credentials []byte, _ ...string) error {
+	m := map[string]string{}
+	if err := json.Unmarshal(credentials, &m); err != nil {
+		return err
+	}
+
```

Comment:
In this implementation, we enter this codepath if the consumer set an `identity` section in provider config, then we unconditionally try to process the kubeconfig even if it can work standalone. In that case, ideally user should not be specifying an `identity` section, but when they do by mistake, this code does not properly handle it. rest See my open issue regarding the similar implementation at provider-helm: https://github.com/crossplane-contrib/provider-helm/issues/206

At this point, I think we should do nil checks for `rc.ExecProvider` and `rc.ExecProvider.Args` otherwise we get nil pointer err below. Then, for the nil check we have two options:
1. since user specified an `identity` section, we assume that they intend to configure an azure ad auth setup. Therefore, we return an error, and say that this kubeconfig is not in expected format (does not have exec provider section)
2. since `execProvider` is not present, we assume that the kubeconfig is a standalone type, and no processing should/can be done here, hence the specified `identity` section is considered superfluous and ignored. We return here with no error. If there are other problems with kubeconfig unrelated to this, let them surface while connecting.

Although not strongly opinionated, I am more inclined towards option 1, as supplying an `identity` section should be considered as an intention to use an azureAD enabled cluster, and the kubeconfig should conform to that, not vice-versa. Even if a working standalone kubeconfig was provided, we should notify user with an error and make sure of their intention, they should either fix their kubeconfig or do not specify any identity configuration.
something like:
```suggestion
if rc.ExecProvider == nil || rc.ExecProvider.Args == nil || len(execProvider.Args) < 1 {
    return errors.New("an identity configuration was specified but the provided kubeconfig does not have execProvider section")
}
```
WDYT?

Reply:
+1 for option 1.

Reply:
implemented